### DISCUSSION TITLE
Format standardization

### DISCUSSION THREAD

**Discussion by mgord9518** [mgord9518/aisap#19] [Announcements]
In this thread, we should discuss the future of aisap not only as a program, but as a standard way to provide AppImage permissions that others may implement themselves, giving users a consistent experience.

aisap currently does the following:
 1. Looks in `~/.config/aisap` for user-provided desktop entries, which contain overridden permissions
 2. Reads from a trusted JSON database in this repo `profiles/profile_database.json`. This is embedded directly into aisap
 3. Looks for permissions in the application bundle's desktop entry that the developer can request

Is this the way forward, or should we consider other options? Can we learn anything from Flatpak, or even re-use its permission-storing scheme for free system integration?

**Comment by xplshn** [comment]
I'm in favor of having sandbox options live at .desktop

**Comment by Samueru-sama** [comment]
Having it the .desktop is good, also maybe looking in `$XDG_DATA_HOME/applications` for a matching `.desktop` that contains permissions would be great.

Btw @mgord9518 I'm holding upstreaming the changes that @xplshn made that fixes the hardcoded XDG vars because right now no sanity check are done on the vars that get returned.

**Reply by xplshn** [reply]
@Samueru-sama no checks have to be done, the `xdg` library always returns a value :)

**Reply by xplshn** [reply]
@mgord9518 the `chains` repo is mostly ready btw. I'm awaiting confirmation by @Samueru-sama on it. And also, I have to catch up with the new features in the zig version

**Reply by mgord9518** [reply]
Awesome! Don't worry about the Zig version's features, It's still completely unusable. I'd like to get back on it but lately I've been busy with other projects.

When I add something new I'll start a discussion so we can remain in parity

**Reply by xplshn** [reply]
> @mgord9518 the `chains` repo is mostly ready btw. I'm awaiting confirmation by @Samueru-sama on it. And also, I have to catch up with the new features in the zig version

I had asked Samu to test it, since I'm not really a user of aisap and all I did was a pretty big reorganization of the code, I swapped some libraries for their std equivalent and also employed the official AppImage library instead

**Reply by mgord9518** [reply]
@xplshn So is CGo required now? Not a huge deal if it is, it's just easier to cross-compile if no system libraries are linked

**Reply by xplshn** [reply]
> @xplshn So is CGo required now? Not a huge deal if it is, it's just easier to cross-compile if no system libraries are linked

Its not needed
